system/src/system_cloud_internal.cpp

```diff
@@ -358,6 +359,7 @@ constexpr const char KEY_RESTORE_EVENT[] = "spark/device/key/restore";
 constexpr const char DEVICE_UPDATES_EVENT[] = "particle/device/updates/";
 constexpr const char FORCED_EVENT[] = "forced";
 constexpr const char UPDATES_PENDING_EVENT[] = "pending";
+constexpr const char KEEPALIVE_INTERVAL_EVENT[] = "particle/device/keepalive";
```

Comment:
The device service presently already has the `[spark/particle]/session/timeout` event, which was made before we had UDP, so the name is now somewhat confusing, but it serves to feed the IdleChecker about how long to consider the device offline.

Reply:
Good to know, thanks! I also noticed that the device service expects the interval to be in seconds, while the current code sends it in milliseconds. Are we sure it's safe if Device OS will start publishing `session/timeout` events? cc @JamesHagerman

Reply:
@m-mcgowan @JamesHagerman Pinging here again. Should I change the event name to `particle/session/timeout`?

Reply:
We still have at least two listeners for that `[spark/particle]/session/timeout` event in the Device Service. Here are the details:

In `reservedEventHandler.js`, we take the data from that event and attempt to store it in MongoDB in the `cores` record under `hardware.session_timeout`.

In `IdleChecker.js` (which is only used by TCP Devices currently) we attempt to pull that record out of mongo when the device connects. If it exists, we use it to control timing around when we call `this.device.disconnect()`.

I queried the production DB using `db.getCollection('cores').find({'hardware.session_timeout': {$exists: true}})` and there are exactly 35 records that contain that key. 

`db.getCollection('cores').find({'hardware.session_timeout': {$gt: 0}})` returns 0 results.

It is safe to say that we are not using that event any more. All of that is legacy

KeepAlive is not related to the session, but rather to the connection itself. At least for UDP devices, the session lives on after a device goes offline.

I think `particle/device/keepalive` is a lot closer to what we're looking for in terms of a namespace for that event. Let's use the version you have in this PR.

Reply:
For visibility: we discussed this internally and decided to use the new event name for now.

Reply:
@m-mcgowan Does the rest of the PR look good to you?